src/Microsoft.ServiceFabric.Services.Remoting/Client/ServiceProxyFactory.cs

```diff
@@ -91,65 +91,45 @@ public ServiceProxyFactory(OperationRetrySettings retrySettings = null)
         {
             var serviceInterfaceType = typeof(TServiceInterface);
 
-#if !DotNetCoreClr
-
             //Use provider to find the stack
-            if (this.proxyFactoryV1 == null && this.proxyFactoryV2 == null)
+            var provider = this.GetProviderAttribute(serviceInterfaceType);
+
+#if !DotNetCoreClr
+            if (provider.RemotingClient.Equals(RemotingClient.V1Client))
             {
-                var provider = this.GetProviderAttribute(serviceInterfaceType);
-                if (provider.RemotingClient.Equals(RemotingClient.V2Client))
-                {
-                    //We are overriding listenerName since using provider we can have multiple listener configured(Compat Mode).
-                    this.overrideListenerName = true;
-                    this.proxyFactoryV2 =
-                        new V2.Client.ServiceProxyFactory(provider.CreateServiceRemotingClientFactoryV2,this.retrySettings);
-                }
-                else
+                // Create V1 proxyFactory
+                if (this.proxyFactoryV1 == null)
                 {
-                    this.proxyFactoryV1 = new V1.Client.ServiceProxyFactory(provider.CreateServiceRemotingClientFactory,this.retrySettings);
+                    this.proxyFactoryV1 =
+                        new V1.Client.ServiceProxyFactory(provider.CreateServiceRemotingClientFactory, this.retrySettings);
                 }
-                
-           }
 
-            if (this.proxyFactoryV1 != null)
-            {
                 return this.proxyFactoryV1.CreateServiceProxy<TServiceInterface>(serviceUri,
                     partitionKey,
                     targetReplicaSelector,
                     listenerName);
             }
-            if (this.overrideListenerName && listenerName==null)
-            {
-                return this.proxyFactoryV2.CreateServiceProxy<TServiceInterface>(serviceUri,
-                    partitionKey,
-                    targetReplicaSelector,
-                    ServiceRemotingProviderAttribute.DefaultV2listenerName);
-            }
-            return this.proxyFactoryV2.CreateServiceProxy<TServiceInterface>(serviceUri,
-                partitionKey,
-                targetReplicaSelector,
-                listenerName);
-#else
-            if (proxyFactoryV2==null)
+#endif
+
+            // Create V2 proxyFactory
+            if (this.proxyFactoryV2 == null)
```

Comment:
Here we default to V2 factory. I thought this was better to assume V2 client rather than doing, if V1 client, else if v2 client, else throw. Because the last branch "should never happen"  - famous last words..

Reply:
Both the scenarios are equally possible, when users just upgrade to new nuget package for existing services V1 is default unless they specify the attribute. Starting with new templates in VS V2 will be made default by specifying the attribute.

Reply:
Seeing as netcore only supports V2, it is more succinct to just assume that the default stack is V2 stack, rather than having another check for the provider. This way the only code that checks the attribute is in non-net-core code. Hope that makes sense?